src/lib/Data/Mapper/ContentUpdateMapper.php

```diff
@@ -38,16 +38,19 @@ public function mapToFormData(ValueObject $contentDraft, array $params = [])
         $data->initialLanguageCode = $languageCode;
 
         $fields = $contentDraft->getFieldsByLanguage($languageCode);
+        $mainLanguageCode = $contentDraft->getVersionInfo()->getContentInfo()->getMainLanguage()->getLanguageCode();
 
         foreach ($params['contentType']->fieldDefinitions as $fieldDef) {
             $isNonTranslatable = $fieldDef->isTranslatable === false;
             $field = $fields[$fieldDef->identifier];
             $data->addFieldData(new FieldData([
                 'fieldDefinition' => $fieldDef,
                 'field' => $field,
-                'value' => $isNonTranslatable && isset($mappedCurrentFields[$fieldDef->identifier])
-                    ? $mappedCurrentFields[$fieldDef->identifier]->value
-                    : $field->value,
+                'value' => $isNonTranslatable
+                    && isset($mappedCurrentFields[$fieldDef->identifier])
+                    && $mainLanguageCode !== $languageCode
+                        ? $mappedCurrentFields[$fieldDef->identifier]->value
+                        : $field->value,
```

Comment:
Please extract these conditions to separate variables with descriptive names. In current form it's highly unreadable.